solutions/cfgym-101102D.mdx

```diff
@@ -0,0 +1,259 @@
+---
+id: cfgym-101102D
+source: CF
+title: Rectangles
+author: Kevin Sheng
+---
+
+Let's first simplify this problem, and then apply the solution of that problem to this one.
+
+Say we had a matrix of all `0`'s and `1`'s. How many submatrices of all `1`'s are there?
+
+We first define a 2D array, $\texttt{longest\_consec}$, that contains for each cell how many `1`'s there are to the right.  
```

Comment:
```suggestion
We first define a 2D array, $\texttt{longest\_consec}$, that contains for each cell how many consecutive `1`'s there are to the left.  
```
I'm not sure, but according to the tables as well as the implementation below, shouldn't it be left instead of right?

Also on line 83, it's written as right too.

Reply:
no it's to the right, the table explains it i think

Reply:
i'm a bit confused here as well. 
![image](https://user-images.githubusercontent.com/8647920/141039204-69849f92-6d12-43fd-84ba-882862e93d44.png)
why would the top right corner be 2 if there's nothing to the right of it?

Reply:
oh wait yeah